### PR TITLE
Implement memory persistence store.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/full_client.toit [hunk 1]
```diff
@@ -325,11 +325,48 @@ interface PersistenceStore:
...
 /**
 A persistence store that stores the packets in memory.
 */
 class MemoryPersistenceStore implements PersistenceStore:
-  UNIMPLEMENTED ::= "UNIMPLEMENTED"
+  storage_ /Map := {:}
+  id_ /int := 0
+
+  store topic/string payload/ByteArray --retain/bool -> int:
+    id := id_++
```
Comment (kasperl) [inline src/full_client.toit:343]: It would be nice with something that doesn't end up generating huge ids.

File: src/full_client.toit [hunk 2]
```diff
@@ -325,11 +325,48 @@ interface PersistenceStore:
...
+    if stored:
+      block.call stored.topic stored.payload stored.retain
+    else:
+      if_absent.call persistent_id
+
+  remove persistent_id/int -> none:
+    storage_.remove persistent_id
+
+  /**
+  Calls the given block for each stored packet.
```
Comment (kasperl) [inline src/full_client.toit:358]: I assume it is important that these are called in insertion order. That should probably be mentioned.

File: src/full_client.toit [hunk 3]
```diff
@@ -325,11 +325,48 @@ interface PersistenceStore:
...
-  UNIMPLEMENTED ::= "UNIMPLEMENTED"
+  storage_ /Map := {:}
+  id_ /int := 0
+
+  store topic/string payload/ByteArray --retain/bool -> int:
+    id := id_++
+    storage_[id] = PersistentPacket_ topic payload --retain=retain
+    return id
+
+  get persistent_id/int [block] [--if_absent] -> none:
```
Comment (kasperl) [inline src/full_client.toit:347]: This interface is a little bit weird. Also the store/get feels like an odd pair naming-wise.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/full_client.toit (hunk 2 above)
Comment (floitsch) [reply src/full_client.toit]: Added a comment to the PersistenceStore interface.

File: src/full_client.toit (hunk 3 above)
Comment (floitsch) [reply src/full_client.toit]: Changed it, by exposing the PersistedPacket

**Pull request merged**